src/services/codefixes/inferFromUsage.ts

```diff
@@ -201,15 +218,71 @@ namespace ts.codefix {
     }
 
     function annotateJSDocParameters(changes: textChanges.ChangeTracker, sourceFile: SourceFile, parameterInferences: ParameterInference[], program: Program, host: LanguageServiceHost): void {
-        const result = mapDefined(parameterInferences, inference => {
+        // find
+        const paramTags = createMap<JSDocParameterTag>();
+        forEach(parameterInferences, inference => {
             const param = inference.declaration;
             const typeNode = inference.type && getTypeNodeIfAccessible(inference.type, param, program, host);
-            return typeNode && !param.initializer && !getJSDocType(param) ? { ...inference, typeNode } : undefined;
+            if (typeNode && !param.initializer && !getJSDocType(param) && isIdentifier(inference.declaration.name)) {
+                paramTags.set(
+                    inference.declaration.name.escapedText as string,
+                    createJSDocParamTag(typeNode, inference.declaration.name as EntityName, !!inference.isOptional, ""));
+            }
         });
-        changes.tryInsertJSDocParameters(sourceFile, result);
+
+        // resolve
+        const signature = parameterInferences.length && parameterInferences[0].declaration.parent;

```

Comment:
Think this should be moved to the top and `if (!signature) return;` before creating the map.

Reply:
Yes indeed.